development/external-plugins/test-untrusted/cmd/main.go

```diff
@@ -39,19 +38,34 @@ func EventHandler(server *externalplugin.Plugin, event externalplugin.Event) {
 		"pr-sender", pr.Sender.Login,
 	)
 	switch pr.Action {
-	case github.PullRequestActionOpened:
+	case github.PullRequestActionOpened, github.PullRequestActionReopened, github.PullRequestActionSynchronize, github.PullRequestActionEdited:
 		pr.GUID = event.EventGUID
 		if pr.Sender.Login == "dependabot[bot]" || pr.Sender.Login == "neighbors-dev-bot" {
+			// If pr action is edited, we check if pr base ref changed. If that's true pr must be retested again.
+			if pr.Action == github.PullRequestActionEdited {
+				l.Info("Received pull request edited action.")
+				var changes struct {
+					Base struct {
+						Ref struct {
+							From string `json:"from"`
+						} `json:"ref"`
+						Sha struct {
+							From string `json:"from"`
+						} `json:"sha"`
+					} `json:"base"`
+				}
+				if err := json.Unmarshal(pr.Changes, &changes); err != nil {
+					l.Info("Failed unmarshal pr event, ignoring.")
+				} else if changes.Base.Ref.From == "" || changes.Base.Sha.From != "" {
+					// the base of the PR did not changed, nothing to do.
+					l.Info("PR base not changed, ignoring.")
+					break
+				}
+			}
 			l.Info("Received pull request event for supported user.")
 			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 			defer cancel()
-			err := ghClient.AddLabelWithContext(ctx, pr.Repo.Owner.Login, pr.Repo.Name, pr.Number, "ok-to-test")
-			if err != nil {
-				l.Errorw("Failed label PR.", "error", err.Error())
-			} else {
-				l.Info("Labeled pr as trusted.")
-			}
-			err = ghClient.CreateCommentWithContext(ctx, pr.Repo.Owner.Login, pr.Repo.Name, pr.Number, "/test all")
+			err := ghClient.CreateCommentWithContext(ctx, pr.Repo.Owner.Login, pr.Repo.Name, pr.Number, "/test all")
```

Comment:
Why use `Context` here? By default the request will timeout after 5 minutes. Use `CreateComment()` instead.

Reply:
Where is this default timeout set?

Reply:
https://github.com/kubernetes/test-infra/blob/7b86cde1b8cfced9f294aac91db753e48d0fd49d/prow/github/client.go#L365